### PR TITLE
add printing gauge field for debugging

### PR DESCRIPTION
**Pull request by jcosborn**
add printing gauge field for debugging
initialize `dslash_type_precondition`
fix CMake to use `QUDA_DIRAC_LAPLACE` instead of old `QUDA_LAPLACE`

### CODE REVIEW

File: lib/color_spinor_util.in.cu [hunk 1]
```diff
@@ -340,7 +340,7 @@ namespace quda {
   {
     if (a.isNative()) {
       constexpr auto order = colorspinor::getNative<Float>(nSpin);
-      print_vector(FieldOrderCB<double, nSpin, nColor, 1, order, Float, Float, false, true>(a), parity, x_cb);
+      print_vector(FieldOrderCB<double, nSpin, nColor, 1, order, Float, Float, false, false>(a), parity, x_cb);
```
Comment (maddyscientist) [inline lib/color_spinor_util.in.cu]: So this last template parameter determines whether we're doing block float or not.  Setting this to false will mean we can print half precision globally normalized fields but not block normalized fields.  What was the use case for setting this to `false`: I think it's only relevant for multigrid null space storage.

**Review by maddyscientist — APPROVED** [review]
Thanks @jcosborn.  Just one query regarding `print_vector` template instantiation, but apart from that this is good to merge.

**Comment by maddyscientist** [comment]
cscs-ci run

File: lib/color_spinor_util.in.cu (hunk 1 above)
Comment (jcosborn) [reply lib/color_spinor_util.in.cu]: I was just printing vectors in CG and needed to change it to get correct output.

Comment (maddyscientist) [reply lib/color_spinor_util.in.cu]: For staggered fermions I presume?

Comment (maddyscientist) [reply lib/color_spinor_util.in.cu]: and in half precision or double/single?

Comment (jcosborn) [reply lib/color_spinor_util.in.cu]: Yes, staggered.  I was looking at both half and single vectors and don't recall the output before the change, but my main focus was on half precision.

Comment (maddyscientist) [reply lib/color_spinor_util.in.cu]: Ok, I've reproduced the garbage with half precision staggered.  Other precisions and Wilson is working fine.  Investigating.....

Comment (maddyscientist) [reply lib/color_spinor_util.in.cu]: Bug located, and it was specifically for the case of half-precision staggered.  I'll push the fix to this PR and revert the change to color_spinor_util.in.cu.

Comment (maddyscientist) [reply lib/color_spinor_util.in.cu]: [bff3830](https://github.com/lattice/quda/pull/1516/commits/bff3830b2efa1d2a67567855179c00cb11640f3a) fixes the issue.  With that now done, we can merge this PR once CI completes.

**Pull request merged**